### PR TITLE
Dash oil and gas

### PR DESCRIPTION
**Pull request by shammamah**
Issue for app: #12 

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL: https://dash-playground.plotly.host/dash-oil-and-gas/
- Current gallery app URL: https://dash-gallery.plotly.host/dash-oil-and-gas/

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dash-oil-and-gas/app.py [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+    dff = df[
+        df["Well_Status"].isin(well_statuses)
+        & df["Well_Type"].isin(well_types)
+        & (df["Date_Well_Completed"] > dt.datetime(year_slider[0], 1, 1))
+        & (df["Date_Well_Completed"] < dt.datetime(year_slider[1], 1, 1))
+    ]
+    return dff
+
+
+def fetch_individual(api):
```
Comment (chriskfwoo) [inline apps/dash-oil-and-gas/app.py]: Just a slight confuse on this function. 
`fetch` naming usually entails you are doing a HTTP request. 
`api` naming is unclear what you are passing through.

File: apps/dash-oil-and-gas/points.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# In[]:
```
[the remark this answers is not in the document]
Comment (celinehuang) [reply apps/dash-oil-and-gas/points.py]: Not quite sure, but removed it and all functionality is still there :)

File: apps/dash-oil-and-gas/app.py [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+
+@app.callback(
+    Output("well_statuses", "value"), [Input("well_status_selector", "value")]
+)
+def display_status(selector):
+    if selector == "all":
+        return list(WELL_STATUSES.keys())
+    elif selector == "active":
+        return ["AC"]
+    else:
```
[the remark this answers is not in the document]
Comment (chriskfwoo) [reply apps/dash-oil-and-gas/app.py]: Is there a reason why you removed the `return []`. I meant you could remove the `else` keyword. So, kinda like this 😄 

```Python
if statement:
   return
return
```

File: apps/dash-oil-and-gas/app.py [hunk 4]
```diff
@@ -0,0 +1,715 @@
...
+
+# Radio -> multi
+@app.callback(Output("well_types", "value"), [Input("well_type_selector", "value")])
+def display_type(selector):
+    if selector == "all":
+        return list(WELL_TYPES.keys())
+    elif selector == "productive":
+        return ["GD", "GE", "GW", "IG", "IW", "OD", "OE", "OW"]
+    else:
+        return []
```
[the remark this answers is not in the document]
Comment (chriskfwoo) [reply apps/dash-oil-and-gas/app.py]: Same thing as above^

**Pull request merged**